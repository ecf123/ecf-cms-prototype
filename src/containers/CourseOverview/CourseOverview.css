.course-overview {
  display: grid;
  grid-template-columns: 3fr 2fr;
  height: 100%;
  gap: 50px;
}
.course-overview__main {
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  margin-left: 32px;
}
.course-overview__side {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  margin-right: 10px;
}