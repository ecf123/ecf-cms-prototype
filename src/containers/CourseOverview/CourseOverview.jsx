import React from 'react'
import DropDown from '../../components/DropDown/DropDown'
import MoreOptions from '../../components/MoreOptions/MoreOptions'
import PathwayIntroduction from '../../components/PathwayIntroduction/PathwayIntroduction'
import PotentialCareers from '../../components/PotentialCareers/PotentialCareers'
import SalaryInfo from '../../components/SalaryInfo/SalaryInfo'

const CourseOverview = () => {
  return (
    <>
        <div className="course-overview__main">
        <PathwayIntroduction image={image} description={"Lorem Ipsum"}/>
        {/* add in pathway component here */}
        <DropDown topicTitle={"Financial Forest"} lessonArr={lessonArr}/>
        </div>
        <div className='course-overview__side'>
        <MoreOptions title="Financal Forest" description="Lorem ipsum" handleDelete={handleDelete} handleEdit={handleEdit}/>
        <SalaryInfo salariesObj={salariesObj}/>
        <PotentialCareers potentialCareers={PotentialCareers}/>
        </div>
    </>
  )
}

export default CourseOverview