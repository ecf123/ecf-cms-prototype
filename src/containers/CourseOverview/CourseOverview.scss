@use "../../styles/main.scss" as *;
@use "../../styles/variables" as *;

.course-overview {
  display: grid;
  grid-template-columns: 3fr 2fr;
  height: 100%;
  gap: 1.875rem;

  &__main {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-left: 2rem;

    &--drop {
      display: flex;
      flex-direction: column;
    }
  }
  &__side {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 0.625rem;
    margin-right: 0.625rem;
    padding-top: 0.625rem;
    background-color: $color-white-smoke;
  }
}
