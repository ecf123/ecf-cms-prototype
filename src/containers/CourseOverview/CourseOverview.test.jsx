import { customRender } from "../../utils/testUtils";
import CourseOverview from "./CourseOverview";

it("Should render the course overview container", () => {
  const { container } = customRender(<CourseOverview />);
  expect(container).toMatchSnapshot();
});
