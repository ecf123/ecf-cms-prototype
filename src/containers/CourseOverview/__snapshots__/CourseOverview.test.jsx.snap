// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the course overview container 1`] = `
<div>
  <section
    class="course-overview"
  >
    <div
      class="course-overview__main"
    >
      <div
        class="pathway-introduction"
      >
        <img
          alt="pathway thumbnail"
          class="pathway-introduction__image"
          src="money-growing-on-tree.svg"
        />
        <h2
          class="pathway-introduction__title"
        >
          Pathway Overview
        </h2>
        <p
          class="pathway-introduction__description"
        >
          Lorem ipsum dolor sit amet. Ea nisi accusamus 33 Quis quidem ut quia obcaecati ex modi dolore 33 suscipit labore. Ut aliquid molestias aut voluptate incidunt nam assumenda quis et iste vero sit dolor dicta At iusto sunt suscipit nulla.
        </p>
      </div>
      <div
        class="pathway-header"
      >
        <h2
          class="pathway-header__title"
        >
          Pathway Structure
        </h2>
        <a
          class="pathway-header__link"
          href="/link"
        >
          <button
            class="button black round-border medium"
          >
            <div
              class="medium-img white-svg"
            >
              <svg>
                pen-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text black"
            >
              Add Course
            </p>
          </button>
        </a>
      </div>
      <div
        class="course-overview__main--drop"
      >
        <nav
          class="drop-down"
        >
          <input
            id="An Introduction to Interest"
            type="checkbox"
          />
          <label
            for="An Introduction to Interest"
          >
            <span
              class="drop-down__title"
            >
              An Introduction to Interest
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="Commercial Banking"
            type="checkbox"
          />
          <label
            for="Commercial Banking"
          >
            <span
              class="drop-down__title"
            >
              Commercial Banking
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="An Introduction to Investing"
            type="checkbox"
          />
          <label
            for="An Introduction to Investing"
          >
            <span
              class="drop-down__title"
            >
              An Introduction to Investing
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="An Introduction to Capital"
            type="checkbox"
          />
          <label
            for="An Introduction to Capital"
          >
            <span
              class="drop-down__title"
            >
              An Introduction to Capital
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="An Introduction to Accounting"
            type="checkbox"
          />
          <label
            for="An Introduction to Accounting"
          >
            <span
              class="drop-down__title"
            >
              An Introduction to Accounting
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="Welcome to HSBC"
            type="checkbox"
          />
          <label
            for="Welcome to HSBC"
          >
            <span
              class="drop-down__title"
            >
              Welcome to HSBC
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
        <nav
          class="drop-down"
        >
          <input
            id="An Introduction to E.C.F."
            type="checkbox"
          />
          <label
            for="An Introduction to E.C.F."
          >
            <span
              class="drop-down__title"
            >
              An Introduction to E.C.F.
              <img
                alt="Topic Arrow"
                class="drop-down__arrow"
                src="upside-down-arrow-grey.svg"
              />
            </span>
          </label>
          <ul
            class="drop-down__list"
          >
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  The Role of Banking and the Financial System
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Currency Risk Management
                </p>
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <p
                  class="drop-down__content"
                >
                  Equity, Assets & Liabilities
                </p>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="course-overview__side"
    >
      <div
        class="more-options"
      >
        <h3
          class="more-options__title"
        >
          Financial Forest
           Options
        </h3>
        <p
          class="more-options__description"
        >
          Nam assumenda quis et iste vero sit dolor dicta At iusto sunt suscipit nulla.
        </p>
        <div
          class="more-options__buttons"
        >
          <button
            class="button light-grey round-border medium"
          >
            <div
              class="medium-img black-svg"
            >
              <svg>
                pen-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text light-grey"
            >
              Edit
            </p>
          </button>
          <button
            class="button red round-border medium"
          >
            <div
              class="medium-img white-svg"
            >
              <svg>
                bin-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text red"
            >
              Delete
            </p>
          </button>
        </div>
      </div>
      <div
        class="salary"
      >
        <h2
          class="salary__heading"
        >
          Average Salaries
        </h2>
        <h5
          class="salary__number"
        >
          £
          70,000
        </h5>
        <span
          class="salary__line"
        />
        <div
          class="salary__jobs"
        >
          <div
            class="salary__titles"
          >
            <h6
              class="salary__title"
            >
              Fund Manager
            </h6>
            <h6
              class="salary__title"
            >
              Financial Analyst
            </h6>
            <h6
              class="salary__title"
            >
              Stock Broker
            </h6>
          </div>
          <div
            class="salary__pays"
          >
            <h6
              class="salary__pay"
            >
              £
              100,000
            </h6>
            <h6
              class="salary__pay"
            >
              £
              60,000
            </h6>
            <h6
              class="salary__pay"
            >
              £
              50,000
            </h6>
          </div>
        </div>
      </div>
      <div
        class="career"
      >
        <h5
          class="career__heading"
        >
          Potential Careers
        </h5>
        <ul
          class="career__list"
        >
          <li
            class="career__item"
          >
            Financial Analyst
          </li>
          <li
            class="career__item"
          >
            Accountant
          </li>
          <li
            class="career__item"
          >
            Investment Analyst
          </li>
          <li
            class="career__item"
          >
            Stock Broker
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
